scripts/west_commands/ncs_ironside_se_update.py

```diff
@@ -0,0 +1,168 @@
+    "F0000003": "StaleFW",
+    "F0000005": "VerifyFailure",
+    "F0000006": "VerifyOK",
+    "F0000007": "UROTUpdateDisabled",
+    "F0000008": "UROTActivated",
+    "F0000009": "RecoveryActivated",
+    "F000000A": "RecoveryUpdateDisabled",
+    "F100000A": "AROTRecovery",
+}
+
+
+class NcsIRONsideSEUpdate(WestCommand):
+    def __init__(self):
+        super().__init__(
+            name="ncs-ironside-se-update",
+            help="NCS IRONside SE update",
+            description="Update IRONside SE.",
+        )
+
+    def do_add_parser(self, parser_adder):
+        parser = parser_adder.add_parser(
+            self.name, help=self.help, description=self.description
+        )
+
+        parser.add_argument(
+            "--zip",
+            help="Path to IRONside SE release ZIP",
+            type=argparse.FileType(mode="r"),
+        )
+
+        parser.add_argument(
+            "--recovery",
+            help="Update IRONside SE recovery instead of IRONside SE itself",
+            action="store_true",
+        )
+
+        parser.add_argument(
+            "--allow-erase", action="store_true", help="Allow erasing the device"
+        )
+        parser.add_argument("--serial", type=str, help="serial number", default=None)
+        parser.add_argument(
+            "--wait", help="Wait time (in seconds) between resets", default=2
+        )
+
+        return parser
+
+    def _decode_version(self, version_mem_dump: str) -> str:
+        def reverse_byte_order(hex_string):
+            return "".join(
+                [hex_string[i : i + 2] for i in range(0, len(hex_string), 2)][::-1]
+            )
+
+        self.dbg(version_mem_dump)
+        version_words = version_mem_dump.split()[2:6]
+
+        # First word is semantic versioned 8 bit each
+        major, minor, patch, seqnum = struct.unpack(
+            "bbbb", bytes.fromhex(version_words[0])
+        )
+
+        # Remaining words is the extraversion C string
+        extraversion_c = bytes.fromhex(
+            "".join([reverse_byte_order(p) for p in version_words[1:]])
+        )
+        extraversion = c_char_p(extraversion_c).value.decode("utf-8")
+
+        return f"{major}.{minor}.{patch}-{extraversion}+{seqnum}"
+
+    def _decode_status(self, status_mem_dump: str) -> str:
+        self.dbg(status_mem_dump)
+        status = status_mem_dump.split()[2]
+        try:
+            return f"{status} - {UPDATE_STATUS_MSG[status]}"
+        except KeyError:
+            return f"{status} - (unknown)"
+
+    def do_run(self, args, unknown_args):
+        def nrfutil_device(cmd: str) -> str:
+            cmd = f"nrfutil device {cmd} --serial-number {args.serial}"
+            self.dbg(cmd)
+            result = subprocess.run(cmd, shell=True, text=True, capture_output=True)
+            return result.stdout
+
+        def get_version() -> str:
+            address = (
+                IRONSIDE_SE_RECOVERY_VERSION_ADDR
+                if args.recovery
```

Comment:
are the lines mixed up here?

Reply:
Its the python formater `black` that did this. Will double check.